src/groups/general_linear.jl

```diff
@@ -0,0 +1,238 @@
+@doc raw"""
+    GeneralLinear{n,𝔽} <: AbstractGroupManifold{𝔽,MultiplicationOperation}
+
+The general linear group, that is, the group of all invertible matrices in $𝔽^{n×n}$.
+
+The default metric is the left-$\mathrm{GL}(n)$-right-$\mathrm{O}(n)$-invariant metric whose
+inner product is written
+$$⟨X_p,Y_p⟩_p = ⟨p^{-1}X_p,p^{-1}Y_p⟩_\mathrm{F} = ⟨X_e, Y_e⟩_\mathrm{F},$$
+where $X_e = p^{-1}X_p ∈ 𝔤l(n) = T_e \mathrm{GL}(n, 𝔽) = 𝔽^{n×n}$ is the corresponding
+vector in the Lie algebra. In the default implementations, all tangent vectors $X_p$ are
+instead represented with their corresponding Lie algebra vectors.
+"""
+struct GeneralLinear{n,𝔽} <: AbstractGroupManifold{𝔽,MultiplicationOperation} end
+
+GeneralLinear(n, 𝔽::AbstractNumbers = ℝ) = GeneralLinear{n,𝔽}()
+
+@doc raw"""
+    GLInvariantMetric{T<:Real,D<:ActionDirection} <: Metric
+
+If the matrix is normal, then the resulting geodesic at the identity element is equivalent
+to the group exponential map.
```

Comment:
I think you have to reference both in the text to show up?

Also documenting the constructor and the three fields would be a great extension.